sdks/java/io/hbase/src/main/java/org/apache/beam/sdk/io/hbase/HBaseSharedConnection.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.hbase;
+
+import java.io.IOException;
+import java.io.Serializable;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hbase.client.Connection;
+import org.apache.hadoop.hbase.client.ConnectionFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Static connection shared between all threads of a worker. Connectors are not persisted between
+ * worker machines as Connection serialization is not implemented. Each worker will create its own
+ * connection and share it between all its threads.
+ */
+public class HBaseSharedConnection implements Serializable {
+  private static final long serialVersionUID = 5252999807656940415L;
+  private static final Logger LOG = LoggerFactory.getLogger(HBaseSharedConnection.class);
+
+  // Transient connection to be initialized per worker
+  // Wrap Connection in array because static Connection cannot be non-null in beam repo
+  private static Connection[] connection = new Connection[1];
```

Comment:
Should use
```
 private static @MonotonicNonNull Connection connection = null;
```
here, then assign connection in getOrCreate. This resolves checkFramework violation.